lalrpop/src/lr1/codegen/parse_table.rs

```diff
@@ -776,16 +776,15 @@ impl<'ascent, 'grammar, W: Write> CodeGenerator<'ascent, 'grammar, W, TableDrive
     fn emit_downcast_fn(&mut self, variant_name: &str, variant_ty: TypeRepr) -> io::Result<()> {
         let spanned_symbol_type = self.spanned_symbol_type();
 
-        rust!(self.out, "fn {}pop_{}<", self.prefix, variant_name);
-        for type_parameter in &self.custom.symbol_type_params {
-            rust!(self.out, "  {},", type_parameter);
-        }
-        rust!(self.out, ">(");
-        rust!(self.out,
-              "{}symbols: &mut ::std::vec::Vec<{}>",
-              self.prefix,
-              spanned_symbol_type);
-        rust!(self.out, ") -> {} {{", self.types.spanned_type(variant_ty));
+        try!(self.out.write_fn_header(&self.grammar,
```

Comment:
Hmm. So I think I avoided using  `write_fn_header` in the first place because it threads all the grammar parameter values through as arguments, which in the case of the `pop` function are not needed. Still, it avoids duplicating some work, so maybe it's a good idea.

But we'll have to change the call site to include those implicit parameters.

(For context, a grammar may do things like `grammar(x: u32)` and then `x` is available in all the actions)